### PR TITLE
FS-2625-2626-mark-as-complete-banner-appears-when-assessment-is-flagged-or-stopped

### PR DESCRIPTION
**Pull request by ramsharma-prog**
This pull request fixes the problem where the assessment complete banner is displayed when the application is flagged or stopped.

### CODE REVIEW

**Review by Adam-W1 — DISMISSED** [review]

File: app/assess/templates/assessor_tasklist.html [hunk 1]
```diff
@@ -49,7 +49,7 @@ <h2 class="assessment-alert__heading govuk-heading-l govuk-!-margin-top-4" autof
             </div>
         </div>
 
-    {% elif sub_criteria_status_completed and current_user_role != "COMMENTER" and not flag.is_qa_complete %}
+        {% elif flag.flag_type.name=="RESOLVED" and sub_criteria_status_completed and current_user_role != "COMMENTER" and not flag.is_qa_complete %}
```
Comment (Adam-W1) [inline app/assess/templates/assessor_tasklist.html:52]: We could probs add a test to check for this condition

Comment (ramsharma-prog) [reply app/assess/templates/assessor_tasklist.html]: Added

**Review by tferns — APPROVED** [review]

File: app/assess/templates/assessor_tasklist.html [hunk 2]
```diff
@@ -49,36 +50,8 @@
...
-                <h2 class="assessment-alert__heading govuk-heading-l govuk-!-margin-top-4">All sections assessed</h2>
-                <p class="govuk-body">You can now mark the assessment as complete when you're ready.</p>
-                <p class="govuk-body">It may be selected for quality assurance (QA) once you've done this.</p>
-                <button class="govuk-button primary-button" data-module="govuk-button" type="submit">Mark as
-                    complete</button>
-            </div>
-        </div>
-    </div>
-</form>
+{% if sub_criteria_status_completed and current_user_role != "COMMENTER" and not flag.is_qa_complete %}
```
Comment (tferns) [inline app/assess/templates/assessor_tasklist.html:53]: Other than refactoring & tests, is this the only relevant line?

Comment (ramsharma-prog) [reply app/assess/templates/assessor_tasklist.html:53]: Yes however tiny logic updated in that macro

**Review by rayhan15243 — DISMISSED** [review]

File: tests/test_jinja_macros.py [hunk 3]
```diff
@@ -718,3 +719,51 @@ def test_assessment_flag(self, request_ctx):
...
+                "macros/assessment_completion.html", "assessment_complete"
+            ),
+            state=type("State", (), {"workflow_status": "COMPLETED"})(),
+            flag=None,
+            csrf_token=generate_csrf(),
+            application_id=1,
+            current_user_role="LEAD_ASSESSOR",
+        )
+
+        rendered_html = rendered_html.replace("\n", "")
```
Comment (gidsg) [inline tests/test_jinja_macros.py:737]: We could pass the html into beautiful soup which would make it much easier to parse.

Comment (ramsharma-prog) [reply tests/test_jinja_macros.py:737]: Okay 👍

Comment (ramsharma-prog) [reply tests/test_jinja_macros.py]: @gidsg I have parsed it with beautiful soup. Please review it 🙏

**Review by rayhan15243 — APPROVED** [review]

**Review by gio-karunakaran — APPROVED** [review]
LGTM

**Pull request merged**